crates/oci/src/client.rs

```diff
@@ -688,6 +704,68 @@ fn registry_from_input(server: impl AsRef<str>) -> String {
     }
 }
 
+fn infer_annotations(
```

Comment:
I think if we covered this method in a unit test that'd take the pressure off adding higher-level `spin registry push` coverage.

Reply:
@vdice I've added unit tests - I think I have all the cases!